common/flatpak-exports.c

```diff
@@ -277,16 +301,99 @@ flatpak_exports_append_bwrap_args (FlatpakExports *exports,
         }
     }
 
-  if (exports->host_fs != 0)
+  if (exports->host_os != 0)
     {
+      const char *os_bind_mode = "--bind";
+      int i;
+
+      if (exports->host_os == FLATPAK_FILESYSTEM_MODE_READ_ONLY)
+        os_bind_mode = "--ro-bind";
+
       if (g_file_test ("/usr", G_FILE_TEST_IS_DIR))
         flatpak_bwrap_add_args (bwrap,
-                                (exports->host_fs == FLATPAK_FILESYSTEM_MODE_READ_ONLY) ? "--ro-bind" : "--bind",
-                                "/usr", "/run/host/usr", NULL);
+                                os_bind_mode, "/usr", "/run/host/usr", NULL);
+
+      for (i = 0; flatpak_abs_usrmerged_dirs[i] != NULL; i++)
```

Comment:
In principle we ought to be giving every multilib directory (`libQUAL` in the FHS) the same treatment: `/libx32` for the inadvisable x86-64 ILP32 ABI, `/libhf` for ARM hardfloat, `/libn32`, `/libo32` and `libn64` for MIPS, and so on.

In practice I don't think I care enough, particularly since users of those obscure architectures are probably using Debian, where `/lib` and `/lib64` are sufficient.

If you'd prefer, I could `g_dir_open("/", ...)` and expose every child that either appears in `flatpak_abs_usrmerged_dirs` or *starts with* `lib`?

Reply:
Nah. I don't really care either.